examples/blame_example.ipynb

```diff
@@ -67,212 +61,246 @@
-     ]
-    }
-   ],
+   "execution_count": null,
+   "metadata": {
+    "id": "L0ufxMO8NdvX",
+    "scrolled": true
+   },
+   "outputs": [],
    "source": [
-    "collisions = None\n",
+    "@dataclass\n",
+    "class LogCollision:\n",
+    "    collision_agents: Tuple[int, int]\n",
+    "    start_time: int \n",
+    "    end_time: int\n",
+    "\n",
+    "def compute_pairwise_collisions(agent_states_history,agent_attributes):\n",
+    "    \"\"\"\n",
+    "    Use polygon intersections to check each agent combination whether there is a collision.\n",
+    "    \n",
+    "    Args:\n",
+    "        agent_states: List[List[AgentState]] #List of all current agent states including x and y coordinates and angle.\n",
+    "        agent_attributes: List[AgentAttributes] #List of static attributes of all agents including agent length and width.\n",
+    "\n",
+    "    Returns:\n",
+    "        List[List[int,int]] #List of colliding pairs of agents: \n",
+    "    \"\"\"\n",
+    "    \n",
+    "    collisions_ongoing = {}\n",
+    "    collisions_all = []\n",
+    "    \n",
+    "    #Iterate through the entire agent state history to find collisions at each time step\n",
+    "    for t, agent_states in enumerate(agent_states_history):\n",
+    "        num_agents = len(agent_states)\n",
+    "        if not num_agents == len(agent_attributes):\n",
+    "            raise Exception(\"Incorrect number of agents or agent attributes.\")\n",
+    "\n",
+    "        #Transform the points of the agents polygon into the common world coordinate frame\n",
+    "        polygons = [None]*num_agents\n",
+    "        for i, (state, attributes) in enumerate(zip(agent_states,agent_attributes)):\n",
+    "            dx = attributes.length/2\n",
+    "            dy = attributes.width/2\n",
+    "\n",
+    "            origin = np.array([state.center.x, state.center.y])\n",
+    "\n",
+    "            angle = state.orientation\n",
+    "            c, s = np.cos(angle), np.sin(angle)\n",
+    "            \n",
+    "            #Rotation matrix from agent coordinate frame to world coordinate frame\n",
+    "            R = np.array([[c, s],\n",
+    "                          [-s, c]])\n",
+    "            #Stacked vectors of agent polygon points in the agents coordinate frame\n",
+    "            #Points are in order around the polygon\n",
+    "            V = np.array([[dx,dy],[dx,-dy],[-dx,-dy],[-dx,dy]])\n",
+    "            #Apply the rotation into the world frame\n",
+    "            VR = np.matmul(V,R)\n",
+    "            \n",
+    "            #Add the displacement between the world frame and the agent frame\n",
+    "            polygons[i] = origin + VR\n",
+    "\n",
+    "        #Check every combination of of agent polygons\n",
+    "        detected_collisions = []\n",
+    "        polygons = [Polygon([p[0], p[1], p[2], p[3]]) for p in polygons]\n",
```

Comment:
How is it different from `polygons = [Polygon(p) for p in polygons]`?

Reply:
According to the Polygon documentation, this would be the same so I've changed it to be the concise option you've put here.